DJ June 2021 (#160)
xaur/decred-news: journal/202106.md

The [reworked code](https://github.com/decred/dcrd/pull/2656) adds a new package called `stdscript` that supports different script versions (this was hard to support previously and is required for the next consensus upgrade), provides a more ergonomic API for developers (including multisig and atomic swap helpers) and improves separation between "standard" and "consensus" (a common point of confusion for new developers). Additionally, the policy of what scripts are considered "standard" has been further tightened to help ensure the typical scripts occupy less space on-chain and further restrict cases that don't make sense.

> As usual, this large change comes in a series of easy-to-digest commits with informative descriptions, tests, benchmarks, examples, all finished with a thorough new [README](https://github.com/decred/dcrd/blob/master/internal/staging/stdscript/README.md).

Reviewer: I suggest using a pinned commit hash for this link because it's in the staging area which means it's going to move and without a pinned hash this link will break:
```suggestion
As usual, this large change comes in a series of easy-to-digest commits with informative descriptions, tests, benchmarks, examples, all finished with a thorough new [README](https://github.com/davecgh/dcrd/blob/62950c2b8b8d831cacf6f2ec46216e420d2658cc/internal/staging/stdscript/README.md).
```

> As usual, this large change comes in a series of easy-to-digest commits with informative descriptions, tests, benchmarks, examples, all finished with a thorough new [README](https://github.com/decred/dcrd/blob/master/internal/staging/stdscript/README.md).

Reviewer: I suggest using a pinned commit hash for this link because it's in the staging area which means it's going to move and without a pinned hash this link will break:
```suggestion
As usual, this large change comes in a series of easy-to-digest commits with informative descriptions, tests, benchmarks, examples, all finished with a thorough new [README](https://github.com/decred/dcrd/blob/62950c2b8b8d831cacf6f2ec46216e420d2658cc/internal/staging/stdscript/README.md).
```

Author: Good idea, thanks. I pinned one other commit hash just in case.

- approved proposals
- treasury consensus bug discovered
- hashrate drop

}

Contents: { sync }

- [Development](#development)
- [People](#people)
- [Governance](#governance)
- [Network](#network)
- [Ecosystem](#ecosystem)
- [Outreach](#outreach)
- [Events](#events)
- [Media](#media)
- [Discussions](#discussions)
- [Markets](#markets)
- [Relevant External](#relevant-external)

## Treasury bug recap

{ what is not affected, what is affected, what to expect }

- https://blog.decred.org/2021/06/25/Treasury-Expenditure-Policy-Bug/
- https://twitter.com/matheusd_tech/status/1409928455974699013
- https://twitter.com/lukebp_/status/1409929016400822279

## Development

The work reported below has the "merged to master" status unless noted otherwise. It means that the work is completed, reviewed, and integrated into the source code that advanced users can [build and run](https://medium.com/@artikozel/the-decred-node-back-to-the-source-part-one-27d4576e7e1c), but is not yet available in release binaries for regular users.

<a id="dcrd" />

**[dcrd](https://github.com/decred/dcrd)**

The way standard scripts are handled has been reworked to address a number of long-standing issues.

> For a quick background, "[scripts](https://devdocs.decred.org/developer-guides/transactions/txscript/overview/)" are small commands stored inside [transactions](https://devdocs.decred.org/developer-guides/transactions/transaction-format/) that consume "input" coins from past transactions and create new "output" coins for the recipient. "Standard scripts" is a subset of all scripts allowed by the consensus that covers most useful operations (sending funds, staking, multisig, etc) optimized for efficiency and network security. Mainnet nodes will normally [refuse](https://github.com/decred/dcrd/pull/2656#issuecomment-851105968) to accept and relay transactions containing non-standard scripts.

Reviewer: I'd suggest the following modifications since it is only standalone transactions that applies to, not those in a block.

```suggestion
For a quick background, "[scripts](https://devdocs.decred.org/developer-guides/transactions/txscript/overview/)" are small commands stored inside [transactions](https://devdocs.decred.org/developer-guides/transactions/transaction-format/) that consume "input" coins from past transactions and create new "output" coins for the recipient. "Standard scripts" is a subset of all scripts allowed by the consensus that covers most useful operations (sending funds, staking, multisig, etc) optimized for efficiency and network security. Mainnet nodes will normally [refuse](https://github.com/decred/dcrd/pull/2656#issuecomment-851105968) to accept and relay standalone transactions (those that not part of a block) containing non-standard scripts.
```

Author: Good clarification, thanks. Added `are` (`those that are not part...`) and committed manually.

The [reworked code](https://github.com/decred/dcrd/pull/2656) adds a new package called `stdscript` that supports different script versions (this was hard to support previously and is required for the next consensus upgrade), provides a more ergonomic API for developers (including multisig and atomic swap helpers) and improves separation between "standard" and "consensus" (a common point of confusion for new developers). Additionally, the policy of what scripts are considered "standard" has been further tightened to help ensure the typical scripts occupy less space on-chain and further restrict cases that don't make sense.

As usual, this large change comes in a series of easy-to-digest commits with informative descriptions, tests, benchmarks, examples, all finished with a thorough new [README](https://github.com/decred/dcrd/blob/master/internal/staging/stdscript/README.md).

> { couldn't resist writing it up, but have a second look if it's too much text for stdscript }

Reviewer: Seems fine to me since it's a major submission this month.

<a id="politeia" />

**[Politeia](https://github.com/decred/politeia)**

User-facing:

- added a button to access [raw Markdown](https://github.com/decred/politeiagui/issues/2415)
- show status of [proposal credit](https://github.com/decred/politeiagui/pull/2417) payment (gives an idea how soon the credits can be used to submit new proposals)
- show a [banner](https://github.com/decred/politeiagui/pull/2456) when the user is viewing `proposals-archive.decred.org` to reduce confusion when clicking site logo
- improved look of the [Flat mode](https://github.com/decred/politeiagui/pull/2434) button
- added proposal name to email [subjects](https://github.com/decred/politeia/pull/1440)
- ~13 bug fixes

Backend and CLI:

- `politeiavoter` [upgraded](https://github.com/decred/politeia/pull/1235) to the new API released with [v1.0.0](https://github.com/decred/politeia/releases/tag/v1.0.0)
- `politeiavoter` will now [retry](https://github.com/decred/politeia/pull/1438) its votes on server errors
- added [MySQL](https://github.com/decred/politeia/pull/1419) implementation for the user database (the backend uses MySQL because of [tlog](https://github.com/decred/politeia/issues/1392))

>   - { can also comment why we need to migrate from cockroach to mysql }

Reviewer 2: > { can also comment why we need to migrate from cockroach to mysql }

The production site currently uses CockroachDB for the user database.  Getting rid of this dependency requires setting up the new MySQL implemenation then migrating the existing data from CockroachDB to MySQL.

Author: Thanks. I assume the intent is to simply deal with one db and not two. Changed to this text for now:

> added [MySQL](https://github.com/decred/politeia/pull/1419) implementation for the user database. The data will need to be migrated from CockroachDB to remove this dependency. The choice was limited to MySQL by [tlog](https://github.com/decred/politeia/issues/1392).

Reviewer 2: Correct.  Ideally, everthing would be running on postgres, but tlog is the limiting factor.
